File: 2160-minimum-sum-of-four-digit-number-after-splitting-digits/2160-minimum-sum-of-four-digit-number-after-splitting-digits.go
func minimumSum(num int) int {
    var numlist []string
    numStr := strconv.Itoa(num)
    
    for _, n := range numStr {
        numlist = append(numlist, string(n))
    }
    
    sort.Strings(numlist)
    num1, _ := strconv.Atoi(numlist[0]+numlist[2])
    num2, _ := strconv.Atoi(numlist[1]+numlist[3])
    return num1+num2
}

File: 2160-minimum-sum-of-four-digit-number-after-splitting-digits/2160-minimum-sum-of-four-digit-number-after-splitting-digits.py
import itertools 

class Solution(object):
    def minimumSum(self, num):
        """
        :type num: int
        :rtype: int
        """
        minsum = num
        numlist = [s for s in str(num)]
        for nums in list(itertools.permutations(numlist, 4)):
            nums = ''.join(list(nums))
            for i in range(1,4):
                num1, num2 = int(nums[:i]), int(nums[i:])
                if num1 + num2 < minsum:
                    minsum = num1 + num2
        return minsum
